src/message.rs

```diff
@@ -93,34 +99,38 @@ pub trait Message: Debug + Send + Sync {
         Self: Default,
     {
         let mut message = Self::default();
-        message.merge_length_delimited(buf)?;
+        message.merge_length_delimited(buf, &mut DecodeContext::default())?;
         Ok(message)
     }
 
     /// Decodes an instance of the message from a buffer, and merges it into `self`.
     ///
     /// The entire buffer will be consumed.
-    fn merge<B>(&mut self, buf: B) -> Result<(), DecodeError>
+    fn merge<B>(&mut self, buf: B, ctx: &mut DecodeContext) -> Result<(), DecodeError>
     where
         B: IntoBuf,
         Self: Sized,
     {
         let mut buf = buf.into_buf();
         while buf.has_remaining() {
             let (tag, wire_type) = decode_key(&mut buf)?;
-            self.merge_field(tag, wire_type, &mut buf)?;
+            self.merge_field(tag, wire_type, &mut buf, ctx)?;
         }
         Ok(())
     }
 
     /// Decodes a length-delimited instance of the message from buffer, and
     /// merges it into `self`.
-    fn merge_length_delimited<B>(&mut self, buf: B) -> Result<(), DecodeError>
+    fn merge_length_delimited<B>(
+        &mut self,
+        buf: B,
+        ctx: &mut DecodeContext,
```

Comment:
likewise

Reply:
We can change this one, although I thought it might be cleared for all `merge` methods to take a context